e2e/wdio/core/tests/dialog.e2e-spec.ts

```diff
@@ -0,0 +1,676 @@
+            openDialog(playgroundDialog);
+            const handleXLocation = Math.floor(getElementLocation(dialog + resizeHandle, 0, 'x'));
+            const handleYLocation = Math.floor(getElementLocation(dialog + resizeHandle, 0, 'y'));
+
+            clickAndDragElement(handleXLocation + 1, handleYLocation + 1, handleXLocation + 250, handleYLocation + 250);
+
+            expect(getElementSize(dialogContainer, 0, 'width')).toBe(600);
+            expect(getElementSize(dialogContainer, 0, 'height')).toBe(600);
+
+            const newHandleXLocation = Math.floor(getElementLocation(dialog + resizeHandle, 0, 'x'));
+            const newHandleYLocation = Math.floor(getElementLocation(dialog + resizeHandle, 0, 'y'));
+
+            clickAndDragElement(newHandleXLocation + 1, newHandleYLocation + 1, newHandleXLocation - 300, newHandleYLocation - 300);
+
+            expect(getElementSize(dialogContainer, 0, 'width')).toBe(400);
+            expect(getElementSize(dialogContainer, 0, 'height')).toBe(400);
+        });
+    });
+
+    describe('visual regression', function() {
+        const complexExample = 12;
+        const stackedExample = 13;
+
+        beforeEach(() => {
+            refreshPage();
+        }, 1);
+
+        it('should check examples visual regression', () => {
+            dialogPage.saveExampleBaselineScreenshot();
+            expect(dialogPage.compareWithBaseline()).toBeLessThan(3);
+        });
+
+        it('should check each dialog', () => {
+            const dialogCount = getElementArrayLength(dialogExamples + button);
+
+            for (let i = 0; i < dialogCount; i++) {
+                if (i === 3 || i === 4 || i === 5 || i === 6) {
+                    // skip due to dynamic content
+                    continue;
+                }
+                if (i === complexExample) {
+                    click(dialogExamples + button, i);
+                    waitForElDisplayed(dialog);
+                    waitForNotDisplayed(busyIndicator);
+                    waitForElDisplayed(dialogItems);
+                    saveElementScreenshot(dialogContainer, `dialog-${i}-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder());
+                    expect(checkElementScreenshot(dialogContainer, `dialog-${i}-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder()))
+                        .toBeLessThan(5, `dialog ${i} screenshot doesn't match baseline`);
+                    click(dialog + button, 2);
+                    continue;
+                }
+                if (i === stackedExample) {
+                    click(dialogExamples + button, i);
+                    waitForElDisplayed(dialog);
+                    saveElementScreenshot(dialogContainer, `dialog-${i}a-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder());
+                    expect(checkElementScreenshot(dialogContainer, `dialog-${i}a-${getImageTagBrowserPlatform()}-`, dialogPage.getScreenshotFolder()))
```

Comment:
what is a and b  is for ?

Reply:
There is a dialog within a dialog. So you open the first (a) like all other dialogs, then in this dialog you click an action to open another dialog (b) on top of it. So, there are 2 dialogs in the same dialog example, so I needed to add a/b so there would be a screen of both.